app/controllers/concerns/event_filters.rb

```diff
@@ -18,9 +18,12 @@ def locations_list(events)
       unless event.building.nil?
         event.building.label
       else
-        event.external_building
+        unless event.external_building.nil?
+          event.external_building
+        end
       end }
 
+    locations.reject! { |s| s.nil? }
```

Comment:
`locations.compact` will eliminate `nil`s more efficiently.

You could just add it into the next line as `locations.compact.uniq.sort`

Reply:
nice